core/src/main/java/org/dozer/MappingProcessor.java

```diff
@@ -336,20 +336,29 @@ private void mapField(FieldMap fieldMapping, Object srcObj, Object destObj) {
       // custom field mapper returns false(indicating the
       // field was not actually mapped by the custom field mapper), proceed as
       // normal(use Dozer to map the field)
-      srcFieldValue = fieldMapping.getSrcFieldValue(srcObj);
-      boolean fieldMapped = false;
-      if (customFieldMapper != null) {
-        fieldMapped = customFieldMapper.mapField(srcObj, destObj, srcFieldValue, fieldMapping.getClassMap(), fieldMapping);
+
+      boolean canMapField = true;
+      if (customFieldMapper != null && customFieldMapper instanceof CustomFieldMapperAndValidator) {
```

Comment:
I do not believe that `customFieldMapper instanceof CustomFieldMapperAndValidator` is the best way to go here. I would say that "mapping the filed" and "filtering source fields" are two different responsibilities. They should be represented in `MapperProcessor` via two different objects of two different interfaces.